src/components/statusBar/StatusBar.svelte

```diff
@@ -4,6 +4,10 @@
   import CircleTick from '~/assets/circleTick.svg';
   import CircleExclamation from '~/assets/circleExclamation.svg';
   import { statusBarStore, settingsStore } from '~/store';
+  import moment from 'moment';
+
+  let lastSyncText = '';
+  setInterval(() => lastSyncText = `Last sync ${moment($statusBarStore.lastSyncDate).fromNow()}.`, 3000);
```

Comment:
I _believe_ that simply re-rendering the status bar text from the store will show the correct time. [The store][1] is currently the source of truth for what information should be shown to the user.

So all that you might need to do (without testing this idea) is to simply instruct the component to re-read data from the store:

```js
let lastSyncText = $statusBarStore.text;

setInterval(() => {
  lastSyncText = $statusBarStore.text
}, 3000);
```

[1]: https://github.com/hadynz/obsidian-kindle-plugin/blob/18a556be9c2f7c4937fb0b4e88898c14a0d105f9/src/store/statusBar.ts#L20-L22

Reply:
I think users might need multi-second accuracy on when sync last happened. So updated the text every minute might be good enough.

Reply:
Interestingly, the value in the derived status bar store is never updated. Setting the interval here works and the text is updated but the value never changes. 

I'm not super familiar with Svelte; is there a way to force re-derivation of the store periodically?

Reply:
Sorry if I sent you on a goose chase without enough guidance.

This is a [really good example][1] from the Svelte docs that resembles what we want to do. Basically "time" is wrapped in a special store object whose value can be set externally using [`readable][2]. This is where we can configure how the timer should behave. This store object can then be passed in the existing `derived` method as a parameter.

Let me know how you go. You are very close!

[1]: https://svelte.dev/tutorial/derived-stores
[2]: https://svelte.dev/docs#readable
[3]: https://github.com/hadynz/obsidian-kindle-plugin/blob/18a556be9c2f7c4937fb0b4e88898c14a0d105f9/src/store/statusBar.ts#L12-L14

Reply:
Okay I've refactored this a bit to get a little closer to what you're hopefully looking for! I've created a store like you mentioned to hold the time, and I've added a derived store in `syncSession.ts` in order to handle the text formatting. 

This works for me locally! Please let me know what you'd like to see changed.

Reply:
FWIW I've continued to use this fork locally and it's been stable and the timing is correct! Please let me know if you'd like me to change or close this PR in some way.